### PR TITLE
api: provide api support

### PR DESCRIPTION
**Pull request by jstar1**
API support is necessary for handling actions
between system controller and stations. There is
still more to be done here, but this is fair start

### CODE REVIEW

**Comment by jstar1** [comment]
Looking for feed back.

File: routes/api.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+// log request
+router.use(function(req, res, next) {
+    console.log(req.method, req.url);
+    next(); 
+});
+
+//TODO: Check if a user is permitted to use station 
+router.route('/user-access')
+.post(function(err, req, res, next) {
```
Comment (danieleynis) [inline routes/api.js]: out of curiosity is there any reason that you did `.route` call first to set the route and then `.post`? As opposed to just doing the `.post` call which can include the route and callback function?

**Comment by danieleynis** [comment]
Seems like there is a conflict and would need to rebase and resolve. I'll take a look again once this is done.

File: routes/api.js (hunk 1 above)
Comment (jstar1) [reply routes/api.js]: @danieleynis just following documentation  __*http://expressjs.com/en/api.html#router.route*__

>the following code shows how to use router.route() to specify various HTTP method handlers.

```
router.route('/users/:user_id')
.get(function(req, res, next) {
  res.json(req.user);
})
.put(function(req, res, next) {
  // just an example of maybe updating the user
  req.user.name = req.params.name;
  // save user ... etc
  res.json(req.user);
})
.post(function(req, res, next) {
  next(new Error('not implemented'));
})
.delete(function(req, res, next) {
  next(new Error('not implemented'));
});```

**Comment by danieleynis** [comment]
Would you be able to add mocha tests to make sure the routes work and return information as intended? Although I am not sure if you intend to do this instead with the hardware sim using python @mongolsamurai made under the `test` folder. We should figure out what we want to do.

**Comment by jstar1** [comment]
@danieleynis I'll check in with @mongolsamurai to run the test he provided.

File: routes/api.js (hunk 1 above)
Comment (jstar1) [reply routes/api.js]: @danieleynis since I'm not currently chaining I'll change it to ```router.post('/route', function ...)

Comment (danieleynis) [reply routes/api.js]: @jstar1 I think it's fine for now if it works, but up to you.

**Pull request closed without merging**